newscoop/library/Newscoop/Services/TemplatesService.php

```diff
@@ -25,33 +27,95 @@ class TemplatesService
      */
     protected $themesService;
 
-    public function __construct(ThemesService $themesService)
+    /**
+     * Publication Service
+     *
+     * @var PublicationService
+     */
+    protected $publicationService;
+
+    public function __construct(ThemesService $themesService, PublicationService $publicationService)
     {
         $this->smarty = \CampTemplate::singleton();
         $this->smarty->assign('gimme', $this->smarty->context());
         $this->themesService = $themesService;
+        $this->publicationService = $publicationService;
         $this->preconfigureSmarty();
     }
 
-    public function fetchTemplate($file, $params = array())
+    /**
+     * Fetch template with smarty
+     *
+     * @param string  $file     template file path
+     * @param array   $params   array with template parameters
+     * @param integer $lifetime template cache lifetime (default: 1400 seconds)
+     * 
+     * @return string Template output
+     */
+    public function fetchTemplate($file, $params = array(), $lifetime = 1400)
     {
-        $this->assignParameters($params);
-
-        return $this->smarty->fetch($file, null, null, null, false);
+        return $this->renderTemplate($file, $params, $lifetime, false);
     }
 
-    public function renderTemplate($file, $params = array())
+    /**
+     * Render template with smarty (will echo outupt)
+     *
+     * @param string  $file     template file path
+     * @param array   $params   array with template parameters
+     * @param integer $lifetime template cache lifetime (default: 1400 seconds)
+     * @param boolean $render   render or just fetch template (default: true [render])
+     * 
+     * @return string Template output
+     */
+    public function renderTemplate($file, $params = array(), $lifetime = 1400, $render = true)
     {
         $this->assignParameters($params);
+        $this->setLifetime($lifetime);
 
-        return $this->smarty->fetch($file, null, null, null, true);
+        return $this->smarty->fetch($file, sha1(serialize($this->smarty->campsiteVector)), null, null, $render);
```

Comment:
setting second parameter allows for caching this same template with different vector